packages/cactus-common/src/main/typescript/exception/exception-helper.ts

```diff
@@ -0,0 +1,330 @@
+import { RuntimeError } from "run-time-error";
+import stringify from "fast-safe-stringify";
+import { Logger } from "../logging/logger";
+
+export class ExceptionHelper {
+  /**
+   * Fetches the exception-content and submits it to the logger
+   *
+   * @param logger logger created thru the LoggerProvider.getOrCreate()
+   * @param exception as an arbitrary exception of whatever type and format it might be
+   * @param optionalAdditionalContext as optional on top info / context on the exception scenario
+   */
+  public static logException(
+    logger: Logger,
+    exception: unknown,
+    optionalAdditionalContext = "",
+  ): void {
+    const exceptionContent = this.getExceptionContent(
+      exception,
+      optionalAdditionalContext,
+    );
+
+    // log to destination
+    logger.error(...exceptionContent);
+  }
+
+  /**
+   * Creates a new RuntimeError based on the incoming exception and context to preserve the complete hierarchy of exception information
+   *
+   * @param exception as an arbitrary exception of whatever type and format it might be
+   * @param context providing context about the exception scenario
+   * @returns a new RuntimeError created from exception to preserve the complete hierarchy of exception information
+   */
+  public static newRuntimeError(
+    exception: unknown,
+    context: string,
+  ): RuntimeError {
+    const exceptionMessageInfo = this.getExceptionMessageInfo(exception);
+    const exceptionStackInfo = this.getExceptionStackInfo(exception);
+    let content = "";
+    let runtimeError = new RuntimeError(context, exceptionMessageInfo.message);
```

Comment:
@m-courtin AFAICT here you always create a `RuntimeError` that then will ***always*** get overwritten by subsequent assignments down the road in the if conditions below, so I recommend removing the extra work that's being done on this line.

Reply:
For the first topic please see comment above.

Regarding the created RuntimeError the initialisation is now done with an undefined as it will be overwritten as you already mentioned